tests/manager/search/test_external_search.py

```diff
@@ -1559,6 +1560,78 @@ def assert_order(sort_field, order, **filter_kwargs):
             customlist_restriction_sets=[[data.extra_list]],
         )
 
+    def test_lane_priority_level_ordering(
+        self, end_to_end_search_fixture: EndToEndSearchFixture
+    ):
+        fixture = end_to_end_search_fixture
+        expect = fixture.expect_results
+
+        data = self._populate_works(fixture)
+
+        def assert_book_is_in_collection(book, in_collection, not_in_collection):
+            book_collections = [x.collection for x in book.license_pools]
+            assert (
+                in_collection in book_collections
+                and not_in_collection not in book_collections
+            )
+
+        collection1_books = {
+            data.b,
+            data.c,
+            data.a,
+        }
+        collection3_books = {
+            data.d,
+            data.e,
+        }
+
+        # ensure that all collection 1 books are in collection1 and not in collection3
+        for book in collection1_books:
+            assert_book_is_in_collection(book, data.collection1, data.collection3)
+        # ensure that all collection 3 books are in collection1 and not in collection1
+        for book in collection3_books:
+            assert_book_is_in_collection(book, data.collection3, data.collection1)
+
+        assert data.e.license_pools[0].collection
+        # collection 1 has the highest priority
+        data.collection1._set_settings(lane_priority_level=10)
+        data.collection3._set_settings(lane_priority_level=1)
+
+        fixture.populate_search_index()
+        facets = FeaturedFacets(minimum_featured_quality=0, entrypoint_is_default=True)
+
+        # use deterministic ordering
+        facets.random_seed = Filter.DETERMINISTIC
```

Comment:
Do we need to use deterministic ordering here? I thought the lane priority level is supposed to overwhelm any randomness (although, I do note that the factor is down from the original 10 to just 1.

Suggestion: It seems like we want to demonstrate that the randomness will be overwhelmed.

Reply:
I suppose that you're right.  I believe I can remove the determinism without a problem.